admin/services/data-access/pupil-age-reason.data.service.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+const { TYPES } = require('./sql.service')
+const R = require('ramda')
+
+const sqlService = require('./sql.service')
+const pupilAgeReasonDataService = {}
+
+/**
+ * Insert pupil age reason record and link with pupil age reason id
+ * @param {Number} pupilId
+ * @param {String} reason
+ * @returns {Promise<Array>}
+ */
+pupilAgeReasonDataService.sqlInsertPupilAgeReason = async function (pupilId, reason) {
+
+  const sql = `
+    DECLARE @pupilAgeReasonId INT
+    DECLARE @pupilAgeReasonTable TABLE (id INT)
+
+    INSERT INTO ${sqlService.adminSchema}.pupilAgeReason
+    (pupil_id, reason)
+    OUTPUT inserted.id INTO @pupilAgeReasonTable
```

Comment:
You don't need to store a single id in a table, which is usually used for sets.  Why not do something like:
```
INSERT INTO [mtc_admin].[pupilAgeReason] (pupil_id, reason) VALUES (@pupilId, @reason);
UPDATE [mtc_admin].[pupil] SET pupilAgeReason_id = SCOPE_IDENTITY()
WHERE pupil_id = @pupilId;
```

And the SQL should be wrapped in a transaction, please.

Reply:
@jon-shipley yea I should have used `SCOPE_IDENTITY` to simplify this, good point 👍 
It's updated.